File: 2017/test/day_17_test.exs
defmodule AdventOfCode2017.Day17Test do
  use ExUnit.Case

  import AdventOfCode2017.Day17
  import AdventOfCode2017.TestHelper

  test "part1" do
    assert part1("3") == 638
    assert part1(input("test/input/day_17.txt")) == 1173
  end

  test "part2" do
    assert part2(input("test/input/day_17.txt")) == 1930815
  end

  test "parse" do
    assert parse(input("test/input/day_17.txt")) == 304
  end
end

File: 2018/test/day_17_test.rb
require "test_helper"
require "day_17"

class Day17Test < Minitest::Test
  def setup
    @day_17 = Day17.new
  end

  def test_part1
    assert_equal 57, @day_17.part1(sample_input_1)
    assert_equal 41027, @day_17.part1(input("day_17.txt"))
  end

  def test_part2
    assert_equal 29, @day_17.part2(sample_input_1)
    assert_equal 34214, @day_17.part2(input("day_17.txt"))
  end

  def sample_input_1
    "x=495, y=2..7
    y=7, x=495..501
    x=501, y=3..7
    x=498, y=2..4
    x=506, y=1..2
    x=498, y=10..13
    x=504, y=10..13
    y=13, x=498..504"
  end
end
